packages/compass-connections-navigation/src/utils.ts

```diff
@@ -0,0 +1,35 @@
+import { css, spacing } from '@mongodb-js/compass-components';
+
+export const getItemPaddingClass = ({
+  level,
+  isPlaceholder,
+  isLegacy,
+}: {
+  level: number;
+  isPlaceholder?: boolean;
+  isLegacy?: boolean;
+}) => {
+  let paddingLeft = 0;
+  switch (level) {
+    case 1:
+      if (isLegacy) {
+        paddingLeft = spacing[2];
+      }
+      break;
+    case 2:
+      paddingLeft = spacing[3];
+      if (isLegacy) {
+        paddingLeft = spacing[6];
+      }
+      break;
+    case 3:
+      paddingLeft = spacing[3] + spacing[5];
+      break;
+  }
+
+  if (isPlaceholder && (level === 1 || (!isLegacy && level === 2))) {
+    paddingLeft += spacing[2];
+  }
+
+  return css({ paddingLeft });
```

Comment:
We should avoid `css` calls in render, this causes class name re-generation and style tag re-insertion on every render. If we have highly dynamic styles like that, we should use style prop instead, or pre-defined actual classNames outside of function scope and just return generated class names here